wpilibc/Athena/src/DriverStation.cpp

```diff
@@ -331,17 +362,22 @@ bool DriverStation::GetStickButton(uint32_t stick, uint8_t button) {
     wpi_setWPIError(BadJoystickIndex);
     return false;
   }
-
-  if (button > m_joystickButtons[stick].count) {
-    ReportJoystickUnpluggedWarning(
-        "Joystick Button missing, check if all controllers are plugged in");
-    return false;
-  }
   if (button == 0) {
     ReportJoystickUnpluggedError(
-        "Button indexes begin at 1 in WPILib for C++ and Java");
+        "ERROR: Button indexes begin at 1 in WPILib for C++ and Java");
+    return false;
+  }
+  std::unique_lock<priority_mutex> lock(m_joystickDataMutex);
+  if (button > m_joystickButtons[stick].count) {
+    // Unlock early so error printing isn't locked.
+    m_joystickDataMutex.unlock();
```

Comment:
It should be sufficient to simply call lock.unlock() here: http://en.cppreference.com/w/cpp/thread/unique_lock/unlock